### PR TITLE
feature/catch-api-error

### PR DESCRIPTION
**Pull request by modpluz**

### CODE REVIEW

File: src/js/config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "APP": {
-    "api_prefix": "http://127.0.0.1:5000/api",
+    "api_base": "http://127.0.0.1:5100/api",
```
Comment (lottspot) [inline src/js/config.json:3]: ...is changing the parameter name really necessary?

File: src/js/core.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-const _ = require('lodash');
 
-let allCfg = JSON.parse(fs.readFileSync(path.join(__dirname, 'config.json')));
-let nodeEnv = process.env.NODE_ENV || 'development';
+const allCfg = JSON.parse(fs.readFileSync(path.join(__dirname, 'config.json')));
+const nodeEnv = process.env.NODE_ENV || 'development';
 let cfg = allCfg.APP;
 
-if (!_.isUndefined(allCfg[nodeEnv])) cfg = _.merge(cfg, allCfg[nodeEnv]);
+if (!isUndefined(allCfg[nodeEnv])) cfg = merge(cfg, allCfg[nodeEnv]);
```
Comment (lottspot) [inline src/js/core.js:12]: I actually think this change hurts code readability

File: src/js/config.json [hunk 3]
```diff
@@ -9,18 +9,19 @@
         "resource": "races",
         "renderMap": [
           {"template": "racesTemplate", "container": "races-container"}
-        ]
+        ],
+        "showInNavigation": true
```
Comment (lottspot) [inline src/js/config.json:13]: this is a good addition 👍

File: src/js/core.js [hunk 4]
```diff
@@ -86,13 +90,40 @@ function parseHTML (htmlString) {
...
+    if (!isUndefined(page.showInNavigation) && page.showInNavigation === true) {
+      if (page.path && page.displayName) navItems.push({ path: page.path, label: page.displayName });
+    }
   });
 
   return navItems;
+}
+module.exports.buildNavigation = buildNavigation;
+
+module.exports.buildPageElement = (section = null) => {
```
Comment (lottspot) [inline src/js/core.js:107]: So glad these page sections are all templates now

**Review by lottspot — APPROVED** [review]

File: src/js/config.json (hunk 1 above)
Comment (modpluz) [reply src/js/config.json:3]: `api_base`, `api_baseurl` makes more sense as opposed to `api_prefix` IMO.

Comment (lottspot) [reply src/js/config.json:3]: Whatever makes you happy ;)

File: src/js/core.js (hunk 2 above)
Comment (modpluz) [reply src/js/core.js:12]: We are currently using 2 or 3 methods from the lodash library, I don't think we should be importing the entire lodash library for this.

Albeit, I'm a readability first kind of guy, perhaps, we should decide if this is a trade off we are willing to make.

Comment (lottspot) [reply src/js/core.js:12]: I feel semi-strongly about airing on the side of readability

**Pull request merged**